porcupine/plugins/filemanager.py

```diff
@@ -83,20 +115,23 @@ def validate_name(*junk: object) -> None:
         else:
             ok_button.config(state="normal")
 
-    var.trace_add("write", validate_name)
+    overwrite_var.trace_add("write", update_dialog_state)
+    file_name_var.trace_add("write", update_dialog_state)
     entry.bind("<Return>", (lambda event: ok_button.invoke()), add=True)
     entry.bind("<Escape>", (lambda event: cancel_button.invoke()), add=True)
     entry.select_range(0, "end")
     entry.focus()
 
     dialog.wait_window()
+
     return new_path
 
 
 def rename(old_path: Path) -> None:
     new_path = ask_name_for_renaming(old_path)
     if new_path is None:
         return
+    clean_copy_path(old_path)
```

Comment:
This rename function is not the only way how a file can get renamed. You can still use other programs to rename files. This means that `clean_copy_path()` won't work correctly in all cases, and at best it works most of the time.

If I understand correctly, the purpose of `clean_copy_path()` is to prevent being able to paste after `Ctrl+C`-ing a file and then deleting or renaming it, as a paste like that would result in a `FileNotFoundError`. I think it would be better to instead check if `copy_path.exists()` before using `copy_path`. We already check whether `copy_path is not None`, so that would mean replacing every `copy_path is not None` check with `copy_path is not None and copy_path.exists()`. Instead of `exists()`, it might be even better to use `is_file()`.

Reply:
Somehow i didn't think about it D: